core/src/main/scala/swave/core/StreamOps.scala

```diff
@@ -312,6 +312,46 @@ trait StreamOps[A] extends Any { self ⇒
   final def slice(startIndex: Long, length: Long): Repr[A] =
     via(Pipe[A] drop startIndex take length named "slice")
 
+  final def sliding(windowSize: Int, emitSingleEmpty: Boolean = false): Repr[immutable.Seq[A]] =
+    slidingTo[immutable.Seq](windowSize, emitSingleEmpty)
+
+  final def slidingTo[M[+ _]](windowSize: Int, emitSingleEmpty: Boolean = false)(
+      implicit cbf: CanBuildFrom[M[A], A, M[A]]): Repr[M[A]] = {
+
+    val builder = cbf.apply()
+
+    prefixAndTail(windowSize).map {
```

Comment:
Do we really need the `prefixAndTail`?
I think we should be able to get away with a `scan` producing the `RingBuffer` followed by a `map`, relying on the map running synchronously right "within" the `scan`. This will look a bit hacky but should be quite efficient. Since it's purely internal and it will be impossible to introduce another transformation between the `scan` and the `map` it should be fine.